src/activity_actor_definitions.h

```diff
@@ -875,5 +879,56 @@ class move_furniture_activity_actor : public activity_actor
         static std::unique_ptr<activity_actor> deserialize( JsonIn &jsin );
 };
 
+class read_activity_actor : public activity_actor
+{
+    public:
+        enum class book_type : int { normal, martial_art };
+
+
+    private:
+        int moves_total;
+        item_location book;
+        // casting is necessary for serialization
+        int bktype = static_cast<int>( book_type::normal );
+
+        // Read until the learner with this ID gets a level
+        bool continuous;
+        int learner_id;
+
+        // Will return true if activity must be set to null
+        bool player_read( avatar &you );
+        bool player_readma( avatar &you );  // Martial arts book
+        bool npc_read( npc &learner );
+
+    public:
+        read_activity_actor() = default;
+
+        read_activity_actor( int moves, item_location &book, bool continuous = false, int learner_id = -1 )
+            : moves_total( moves ), book( book ),
+              continuous( continuous ), learner_id( learner_id ) {};
+
+        activity_id get_type() const override {
+            return activity_id( "ACT_READ" );
+        }
+
+        static void read_book( Character &learner, const cata::value_ptr<islot_book> &islotbook,
+                               SkillLevel &skill_level, const double penalty );
```

Comment:
```suggestion
                               SkillLevel &skill_level, double penalty );
```

Reply:
I don't modify `penalty` so I don't see why it shouldn't be a const.

Reply:
`penalty` is passed by value, so whether it is const in the function body does not matter to the outside caller. The error messages says exactly this (`const-qualification of parameters only has an effect in function definitions`).

Reply:
Oh. I didn't even see the error message, sorry. Clang is really weird.